src/Input/Item.hs

```diff
@@ -178,7 +178,9 @@ hseToSig = tyForall
             TVar a b -> TVar a (b ++ [ty y])
         ty (TyVar _ x) = TVar (fromName x) []
         ty (TyCon _ x) = TCon (fromQName x) []
+#if MIN_VERSION_haskell_src_exts(1,20,0)
         ty (TyInfix an a (UnpromotedName _ b) c) = ty $ let ap = TyApp an in TyCon an b `ap` a `ap` c
+#endif
```

Comment:
Note this probably isn't the right fix - see https://github.com/ndmitchell/hoogle/commit/c4ee561f2dcec959bbd4bc269fb764ece6b7f259#diff-4cbfe74a8c8cd726976adbaf48d37453 for what I had to do to get it into this state.